lib/matplotlib/testing/decorators.py

```diff
@@ -134,10 +135,15 @@ def copy_baseline(self, baseline, extension):
             # os.symlink errors if the target already exists.
             with contextlib.suppress(OSError):
                 os.remove(expected_fname)
-            try:
-                os.symlink(orig_expected_path, expected_fname)
-            except OSError:  # On Windows, symlink *may* be unavailable.
+            if ('microsoft-standard' in uname().release  # WSL2
+                    or 'Microsoft' in uname().release):  # WSL1
+                # On WSL, symlink breaks silently
                 shutil.copyfile(orig_expected_path, expected_fname)
+            else:
+                try:
+                    os.symlink(orig_expected_path, expected_fname)
+                except OSError:  # On Windows, symlink *may* be unavailable.
+                    shutil.copyfile(orig_expected_path, expected_fname)
```

Comment:
I think you could write this as:
```
try:
    if ...:
        raise OSError
    os.symlink(...)
except OSError:
    shutil.copyfile(...)
```
to avoid writing that `copyfile` twice.